### PR TITLE
Openapi import

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -246,6 +253,7 @@ val commonsIOV = "2.4"
 val dropWizardV = "5.0.0-rc3"
 val scalaCollectionsCompatV = "2.3.2"
 val testcontainersScalaV = "0.39.3"
+val nettyV = "4.1.48.Final"
```
[the remark this answers is not in the document]
Comment (mproch) [reply build.sbt:253]: we have it currently, I just wanted to fix one version. Let's do further upgrade in other PR

File: engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala [hunk 2]
```diff
@@ -1,56 +1,67 @@
...
+import sttp.client.asynchttpclient.future.AsyncHttpClientFutureBackend
 
+import java.net.URL
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.{ExecutionContext, Future}
 import scala.io.Source
-import scala.jdk.CollectionConverters.{asScalaBufferConverter, mapAsScalaMapConverter}
-
-//TODO: move to integration tests or remove "real" RestDB invocation
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala]: done :)

File: demo/docker/docker-compose-env.yml [hunk 3]
```diff
@@ -151,7 +151,9 @@ services:
       - jobmanager
       - taskmanager
       - grafana
-
+  customerservice:
+    container_name: nussknac_customerservice
```
Comment (dswiecki) [inline demo/docker/docker-compose-env.yml:155]: is a nussknac our new shorthand ;)

File: demo/docker/docker-compose.yml [hunk 4]
```diff
@@ -7,14 +7,17 @@ services:
     ports:
       - "3081:8080"
     environment:
-      NUSSKNACKER_CONFIG_FILE: ${NUSSKNACKER_CONFIG_FILE:-/opt/nussknacker/conf/application.conf}
+      #multiple, comma separated, config files can be used. They will be merged in order, via HOCON fallback mechanism
```
Comment (dswiecki) [inline demo/docker/docker-compose.yml:10]: probably should be part of some documentation

File: docs/components/OpenAPI.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+- Each operation is mapped to one enricher
+- If operationId is configured it's used as enricher name, otherwise we create it as concatenation
+of HTTP method, path and parameters (to provide some level of uniqueness)
+- Parameters (path, query of header) are used to define enricher parameters
+- If specification declares body parameter as object, we expand to parameter list  
+- We expect operation to define 200/201 response, returned object is the one that is the result of enricher
+- We map 404 HTTP code to null value of enricher
+- We use `externalDocs.url` to extract documentation link
+
+Table below describes data types that OpenAPI integration handles:
```
Comment (dswiecki) [inline docs/components/OpenAPI.md:18]: I've found greate Intelij Plugin for formatting markdown tables https://plugins.jetbrains.com/plugin/7550-pipe-table-formatter/reviews

File: engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        enricher.open(JobData(metaData, ProcessVersion.empty, DeploymentData.empty), new StandaloneContextPreparer(NoOpMetricsProvider).prepare("1"))
+
+        withFixture(test.toNoArgTest(enricher))
+      }
+    } finally {
+      backend.close()
+    }
+  }
+
+  test("companies") { service =>
```
Comment (dswiecki) [inline engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala:57]: probably the meaning of `companies` has chanaged

File: engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      return new BaseSwaggerEnricherCreator {
+        override def create(rootUrl: Option[URL], swaggerService: SwaggerService, fixedParams: Map[String, () => AnyRef]): BaseSwaggerEnricher =
+          new FlinkSwaggerEnricher(rootUrl, swaggerService, fixedParams, httpClientConfig)
+      }
+    }
+    if (isStandaloneBased) {
+      return new BaseSwaggerEnricherCreator {
+
+        //TODO: figure out how to pass client...
+        lazy val asyncHttpClient = new DefaultAsyncHttpClient(httpClientConfig.toAsyncHttpClientConfig(None).build())
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala:78]: I added comment - maybe we'll want to create it only once?

File: engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    if (isFlinkBased) {
+      return new BaseSwaggerEnricherCreator {
+        override def create(rootUrl: Option[URL], swaggerService: SwaggerService, fixedParams: Map[String, () => AnyRef]): BaseSwaggerEnricher =
+          new FlinkSwaggerEnricher(rootUrl, swaggerService, fixedParams, httpClientConfig)
+      }
+    }
+    if (isStandaloneBased) {
+      return new BaseSwaggerEnricherCreator {
+
+        //TODO: figure out how to pass client...
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala]: I added comment, the method is used 2 lines below :)

File: demo/docker/docker-compose-env.yml (hunk 3 above)
Comment (mproch) [reply demo/docker/docker-compose-env.yml]: :+1:

File: docs/components/OpenAPI.md (hunk 5 above)
Comment (mproch) [reply docs/components/OpenAPI.md:18]: :100: thanks :)

File: engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala (hunk 6 above)
Comment (mproch) [reply engine/components/openapi/src/it/scala/pl/touk/nussknacker/openapi/functional/RestDBServiceSpec.scala]: :+1: I added better description...

File: demo/docker/docker-compose.yml (hunk 4 above)
Comment (mproch) [reply demo/docker/docker-compose.yml:10]: Yes, it will be part of configuration documentation, but I think here it's also ok to have it?

**Review by dswiecki — APPROVED** [review]

**Pull request merged**